### PR TITLE
Disable zlib when building libMLIRMIOPEN

### PR DESCRIPTION
**Pull request by yiqian1**
Disable zlib when building libMLIRMIOPEN to avoid linking errors during the MIOpen build.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,7 @@ if( BUILD_FAT_LIBMLIRMIOPEN )
   set(BUILD_SHARED_LIBS OFF CACHE BOOL "")
   set(LLVM_BUILD_LLVM_DYLIB OFF CACHE BOOL "")
   set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE STRING "")
+  set(LLVM_ENABLE_ZLIB "OFF" CACHE STRING "")
```
Comment (jerryyin) [inline CMakeLists.txt]: Please move it up to around line 16. This shouldn't be used either in the library or driver/runner.

Also, please help add the below back:
https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/b0d60d9e943c3173e6665f35c6b84ebcc1b1c0bc/CMakeLists.txt#L10

**Review by jerryyin — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -15,6 +15,8 @@ set(MLIR_MIOPEN_DRIVER_TEST_GPU_VALIDATION 1 CACHE BOOL "Enable E2E tests with G
 
 set(MLIR_MIOPEN_SQLITE_ENABLED 0 CACHE BOOL "Enable SQLite integration")
 set(MLIR_ENABLE_SQLITE "${MLIR_MIOPEN_SQLITE_ENABLED}")
+set(LLVM_ENABLE_ZLIB "OFF" CACHE STRING "")
```
Comment (jerryyin) [inline CMakeLists.txt:18]:
```suggestion
set(LLVM_ENABLE_ZLIB OFF CACHE STRING "")
```

Comment (jerryyin) [inline CMakeLists.txt:29]:
```suggestion
set(LLVM_ENABLE_ZLIB OFF CACHE BOOL "")
```

Comment (yiqian1) [reply CMakeLists.txt:29]: LLVM_ENABLE_ZLIB is a string variable.

external/llvm-project/llvm/CMakeLists.txt:

set(LLVM_ENABLE_ZLIB "ON" CACHE STRING "Use zlib for compression/decompression if available. Can be ON, OFF, or FORCE_ON")

Comment (jerryyin) [reply CMakeLists.txt:29]: Nvm

**Review by jerryyin — APPROVED** [review]

**Pull request merged**